### PR TITLE
Fix overlapping, streaming reverse, and cleanup

### PR DESCRIPTION
**Pull request by ashenfad**
## Summary
- **Streaming reverse iteration via time-negation** — eliminates O(n) materialization in Intersection, Difference, and Complement by negating timestamps, running the forward-only sweep, and negating outputs
- **Correct `overlapping()` for Difference and Complement** — the base implementation used a narrow `fetch(point, point+1)` window that missed subtractors/source intervals needed for correct results; both now scan the full relevant range
- **Thread-safe `CachedTimeline`** — a `threading.Lock` around evict/fill/stitch in `fetch` prevents concurrent corruption (already possible during I/O, critical for free-threaded Python 3.13+)
- **Immutable `RecurringPattern.exdates`** — changed from mutable `set` to `frozenset` to prevent surprise aliasing mutations
- **Replace defensive `getattr` with direct attribute access** — all guarded attributes are unconditionally set in `__init__`, so `getattr` fallbacks silently masked potential bugs

## Test plan
- [x] All 462 tests pass (439 existing + 23 new)
- [x] New `tests/test_overlapping.py` covers Difference, Complement, nested `~(A-B)`, double difference `(A-B)-C`, `(~A)-B`, boundary conditions, and edge cases

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: calgebra/mutable/memory.py [hunk 1]
```diff
@@ -349,7 +345,9 @@ def _remove_recurring_instance(self, interval: Interval) -> list[WriteResult]:
                         )
                     ]
 
-                pattern.exdates.add(interval.start)
+                # Replace pattern with updated exdates (immutable frozenset)
+                pattern.exdates = pattern.exdates | {interval.start}
+                self._recurring_patterns[i] = (stored_id, pattern)
```
[the remark this answers is not in the document]
Comment (ashenfad) [reply calgebra/mutable/memory.py]: 2dee811
